### ISSUE TITLE
Make specs compatible with jquery v3

### ISSUE THREAD

**Issue by paulfalgout** [marionettejs/backbone.marionette#3095]
We can't currently update the jquery dependency used in our tests because 6 tests are incompatible with jquery3.  Again this isn't Marionette's code, that works fine, but the tests themselves are using deprecated functionality.

```
1) composite view when rendering with a false template should bind the ui hash to jQuery selectors:
     AssertionError: expected undefined to equal '.test-element'
      at Context.<anonymous> (composite-view.spec.js:950:61)

  2) region when initializing a region and passing an "el" option should manage the specified el:
     AssertionError: expected undefined to equal '#foo'
      at Context.<anonymous> (region.spec.js:807:43)

  3) layoutView when defining region selectors using @ui. syntax should apply the relevant @ui. syntax selector to a simple string value:
     AssertionError: expected undefined to equal '.craft'
      at Context.<anonymous> (view.child-views.spec.js:558:64)

  4) layoutView when defining region selectors using @ui. syntax should apply the relevant @ui. syntax selector to selector in a region definition object:
     AssertionError: expected undefined to equal '.bros'
      at Context.<anonymous> (view.child-views.spec.js:562:66)

  5) layoutView when defining region selectors using @ui. syntax should apply the relevant @ui. syntax selector to el in a region definition object:
     AssertionError: expected undefined to equal '.toadstool'
      at Context.<anonymous> (view.child-views.spec.js:566:69)

  6) item view when rendering with a false template should bind the ui hash to jQuery selectors:
     AssertionError: expected undefined to equal '.test-element'
      at Context.<anonymous> (view.spec.js:89:52)
```

**Comment by paulfalgout** [comment]
https://github.com/marionettejs/backbone.marionette/issues/2940 is loosely related

**Comment by rafde** [comment]
Those are all `jQuery#selector` issues from the looks of it.
Would using [`.matches`](http://caniuse.com/#feat=matchesselector) suffice as a fix for all those test?
example
```
expect(this.view.ui.testElement[0].matches('.test-element')).to.be.true
```

**Comment by rafde** [comment]
OK. I'll get to it.

**Comment by rafde** [comment]
>2) region when initializing a region and passing an "el" option should manage the specified el:
AssertionError: expected undefined to equal '#foo' at Context.<anonymous> (region.spec.js:807:43)

This is the only test that you can work around by doing `expect(this.region.el).to.equal(this.el);`

The other test can't use `.matches` because ui values are empty jQuery instances and you can't match against anything in them. Either the test go away since they are trying to test validity through a removed property from a 3rd party library, or the test have to move and test for elements that exist as DOM and match the selector test (not to be confused with `.selector`).

So... should I
1) remove these deprecated test 
or 
2) replace test with DOM match test somewhere else where the DOM elements do exist?

**Comment by paulfalgout** [comment]
I think I'd vote for #1.  I can't think of a situation where the other tests would fail in isolation.  Some other test would catch the same issues.. if one were to arise.

**Comment by rafde** [comment]
Ah right, I'll git to removing them.

**Comment by denar90** [comment]
resolved #3107
